src/libpgmoneta/json.c

```diff
@@ -620,6 +621,23 @@ parse_string(char* str, uint64_t* index, struct json** obj)
          // The key
          while (idx < len && str[idx] != '"')
          {
+            char ec_ch;
+            // handle escape character
+            if (str[idx] == '\\')
+            {
+               if (handle_escape_char(str, &idx, len, &ec_ch))
+               {
+                  goto error;
+               }
+               key = pgmoneta_append_char(key, ec_ch);
+               continue;
+            }
+            // check for illegal use cases
+            if (str[idx] == '\'' || str[idx] == '\"' || str[idx] == '\\')
```

Comment:
This step is necessary in case where we get something like `{"name":"as'hu"}` here `'` is not escaped so will bypass the `if (str[idx] == '\\')` condition and fallback to this check.

`str[idx] == "\""` is redundant however

Reply:
Seems that `'` is not required to be escaped in https://www.json.org/json-en.html. Also it won't cause any ambiguity when we only use double quotes for string enclosement. I think we can allow it. As for `\`, I can't really think of cases where it falls into the `if` condition, insteading of going into `handle_escape_char` and then erroring out. Though feel free to raise counter example. In terms of `"`, yeah I think we've talked about it being redundant.

If none of the case is necessary, we can safely get rid of the check.

Reply:
> Seems that `'` is not required to be escaped in https://www.json.org/json-en.html. Also it won't cause any ambiguity when we only use double quotes for string enclosement. I think we can allow it. As for `\`, I can't really think of cases where it falls into the `if` condition, insteading of going into `handle_escape_char` and then erroring out. Though feel free to raise counter example. In terms of `"`, yeah I think we've talked about it being redundant.
> 
> If none of the case is necessary, we can safely get rid of the check.

Yeah, absolutely correct, We actually don't need that extra check, my bad.